debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/unbuffered/UnbufferedLogMinerStreamingChangeEventSource.java

```diff
@@ -214,6 +214,27 @@ protected void executeDataChangeEventPreDispatchSteps(LogMinerEventRow event) th
         }
     }
 
+    /**
+     * Recomputes the resume position, optionally updating the offsets if the resume position
+     * changes from its prior value.
+     *
+     * @param resumeScn the current resume system change number, should not be {@code null}
+     * @param commitScn the current commit system change number, should not be {@code null}
+     * @return the next iteration's resume system change number, never {@code null}
+     * @throws SQLException if a database exception occurs
+     */
+    private Scn computeResumeScnAndUpdateOffsets(Scn resumeScn, Scn commitScn) throws SQLException {
+        final Scn computedResumeScn = resumePositionProvider.computeResumePositionFromLogs(
+                resumeScn, commitScn, getCurrentLogFiles());
+
+        if (!computedResumeScn.equals(resumeScn)) {
```

Comment:
@Naros Just to make sure, there is no way that `computedResumeScn` can be null? If it is the case then feel free to merge yourself.

Reply:
There is not, but thanks for asking. I had to go back and double check the code to be sure :grin: